docs/designs/ps_design.md

```diff
@@ -1,377 +1,94 @@
 # Parameter Server Design
+This document describes the design of a distributed parameter server (PS) to support training scalability and PS fault tolerance in ElasticDL.
 
 
-## Overview
-Currently, there is one parameter server (PS) co-existed with the master. In order to support PS with multiple nodes and PS fault tolerance, we need to separate PS from the master. Besides a KV store for model parameters, PS should also support updating parameters using gradients. Currently, models have two kinds of parameter, variable and embedding vector. PS should support model variable update and embedding table sparse update.
+## Motivation
+Parameter server based distributed training uses data parallelism to speed up training. PS stores model parameters. There are multiple workers that get model parameters from PS, compute gradients using different training data and send computed gradients to PS. PS iteratively updates these model parameters using gradients sent by workers. A PS based distributed training can use an arbitrary number of workers to support the scalability of training data size.
 
-The master will create a PS with *N* PS nodes, where *N* is specified by the user. Each model variable and embedding vector has a corresponding PS node. Thus, every PS node has a subset of model variables and embedding tables.
+If a model has a very large size, it may not fit in the memory of a single parameter server. For example, many recommending and ranking models use embedding layers. The parameter of an embedding layer is an embedding table, which consists of multiple embedding vectors. When the number of embedding vectors is too large to store in a single PS, we need a distributed PS. A distributed PS contains multiple PS pods.  The distributed PS can partition model parameters and store different partitions in different PS pods. 
```

Comment:
If a model has a very large size, it may not fit in the memory ...

=>

Big models might overrun the memory space of a single parameter server process.

Reply:
I guess the author wanted to express the following logic derivation process in this paragraph:

1. We want to have one or more parameter server instances in each job.
1. A reason is that models could be large and overrun the memory space of each single process.
1. In such cases, we need to shard the model.
1. Even if the model is not too big and fits in the memory of a single parameter server instance,
1. we might still want model sharding,
1. because this distribute the parameter access communication from the workers in the job.
1. and it also distributes the computation like embedding vector combination and parameter optimization over multiple parameter server instances.